sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_resource_configuration.py

```diff
@@ -50,11 +50,18 @@ def as_dict(self) -> Dict[str, Any]:
 
     @classmethod
     def _to_dict(cls, obj: Any) -> Any:
+        # pylint: disable=protected-access
         if isinstance(obj, dict):
+            # if obj has attribute _NEED_KEY_MAPPING, it should be an internal AISuperComputerConfiguration object,
```

Comment:
I'd recommend to make it a function for `BaseProperty` to avoid using `has_attr` judgement

Reply:
Updated as offline sync, create a new base class `PascalCaseProperty` and override its `items` method.

I have not found any other usage, so let's wait for the CI result.